jenkins/tests/Jenkinsfile

```diff
@@ -679,6 +679,17 @@ pipeline {
            }
         }
     }
+
+ 
+ def extractReason(String log) {
+    // Extract reason from the log
+    def reason = ""
+    def reasonPattern = /message="([^"]*)">/
+    def matcher = log =~ reasonPattern
+    if (matcher.find()) {
+        reason = matcher.group(1).trim()
+    }
+    return reason
```

Comment:
If you merge the latest version of dev into this branch, it should remove these changes. This method `extractReason` is already defined in this file , so it shouldn't be here. (If merging dev into this branch doesn't get rid of one of the methods, you can just make sure they are the same and manually remove one.)